packages/engine/lib/error/src/lib.rs

```diff
@@ -228,23 +224,69 @@ pub use self::{
     result::{IntoReport, Result, ResultExt},
 };
 
-/// Trait alias for an error context.
+/// Defines the current context of a [`Report`].
 ///
-/// Note: This is currently defined as trait but will be changed to be a trait alias as soon as
-///   it's stabilized
-// TODO: change to `pub trait Context = ...`
-//   Tracking issue: https://github.com/rust-lang/rust/issues/41517
-pub trait Context: Provider + fmt::Display + fmt::Debug + Send + Sync + 'static {}
-impl<C: Provider + fmt::Display + fmt::Debug + Send + Sync + 'static> Context for C {}
-
-/// Trait alias for a contextual message.
+/// Every [`Error`] also can act as a `Context`. If [`Error`] is not implemented or in a non-std
+/// environment, this trait can be implemented manually. In most cases, implementing this is not
+/// needed.
+///
+/// [`Error`]: std::error::Error
+///
+/// ## Example
+///
+/// Used for creating a [`Report`] or for switching the [`Report`]s context:
```

Comment:
We love English being ambiguous. In this instance it would be possessive, so the context belongs to the Report. 
https://www.thepunctuationguide.com/apostrophe.html